analyzer/block/block.go

```diff
@@ -223,13 +214,40 @@ func (b *blockBasedAnalyzer) Start(ctx context.Context) {
 		// Process blocks.
 		b.logger.Debug("picked blocks for processing", "heights", heights)
 		for _, height := range heights {
+			// If running in slow-sync, we are likely at the tip of the chain and are picking up
+			// blocks that are not yet available. In this case, wait before processing every block,
+			// so that the backoff mechanism can tweak the per-block wait time as needed.
+			//
+			// Note: If the batch size is greater than 50, the time required to process the blocks
+			// in the batch will exceed the current lock expiry of 5min. The analyzer will terminate
+			// the batch early and attempt to refresh the locks for a new batch.
+			if b.slowSync {
+				select {
+				case <-time.After(backoff.Timeout()):
+					// Process the next block
+				case <-batchCtx.Done():
+					b.logger.Info("batch locks expiring; refreshing batch")
```

Comment:
Can `break` here, to avoid going through remaining blocks if the context is canceled anyway.